### ISSUE TITLE
what is "mask_proto_normalize_emulate_roi_pooling" used to do?

### ISSUE THREAD

**Issue by shoutOutYangJie** [dbolya/yolact#136]
if  I want to train model without crop, I need turn off "mask_proto_crop". But I am concerned that It will have an impact on following code.

>  if cfg.mask_proto_normalize_emulate_roi_pooling:  # True
                weight = mask_h * mask_w if cfg.mask_proto_crop else 1   # mask_proto_crop is True
                pos_get_csize = center_size(pos_gt_box_t)
                gt_box_width  = pos_get_csize[:, 2] * mask_w
                gt_box_height = pos_get_csize[:, 3] * mask_h
                pre_loss = pre_loss.sum(dim=(0, 1)) / gt_box_width / gt_box_height * weight

I am afriad that if i change "crop" from True to False, the gridents will become bad.  It seems that the above code is used to balance the grident?

**Comment by dbolya** [comment]
Your concern is valid. If you want to train without cropping, you need to set both
```python
'mask_proto_crop': False,
'mask_proto_normalize_emulate_roi_pooling': False,
```
The latter just emulates how the loss would be weighted *if* we were using repooling instead of cropping (i.e., divide the mask loss by the gt box area instead of the prototype dimensions).

Also, since training without cropping changes how the loss is weighted, you also need to fiddle with the mask alpha. I think the setting we used was
```python
'mask_alpha': 30.6
```
but I can't remember 100%, so try this and if the mask loss is too high / low, change it accordingly.

**Comment by shoutOutYangJie** [comment]
where does your experience on how to define this loss weights from?  It works for my task!  thanks!

**Comment by dbolya** [comment]
Just make sure that the magnitude is similar to the other losses. I didn't really rigorously test it so it may not be optimal ¯\_(ツ)_/¯
